profitbricks/src/test/java/org/jclouds/profitbricks/http/ResponseStatusFromPayloadHttpCommandExecutorServiceTest.java

```diff
@@ -95,4 +100,43 @@ public void testUnauthorized() throws Exception {
          server.shutdown();
       }
    }
+   
+   @Test
+   public void testOverLimitSettings() throws Exception {
+      MockWebServer server = mockWebServer();
+      server.enqueue( new MockResponse().setResponseCode(503).setBody(payloadFromResource( "/fault-413.xml" )));
+      
+      ProfitBricksApi pbApi = api(server.getUrl("/"));
+      ServerApi api = pbApi.serverApi();
+      
+      try {
+         api.createServer( Server.Request.CreatePayload.create( "some-datacenter-id", "node1", 99, 12800) );
+         fail( "Request should have failed.");
+      } catch (Exception ex){
+         assertTrue(ex instanceof InsufficientResourcesException, "Exception should be InsufficientResourcesException");
+      } finally {
+         pbApi.close();
+         server.shutdown();
+      }
+   }
+   
+   @Test
+   public void testServiceUnderMaintenance() throws Exception {
+      MockWebServer server = mockWebServer();
+      for ( int i = 0; i <= MAX_RETRIES; i++ )  // jclouds retries 5 times
```

Comment:
You can set the `Constants.PROPERTY_MAX_RETRIES` property when building the context to a lower value if you don't need all those 5 retries.